File: Anul 2/Semestrul 2/MPP/labs/Tema 5-proiect Server Csharp-Client Java/proiectSCcsharp/angajatPersistence/RepoAngajat.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using proiectSCcsharp.repository.interfaces;
using proiectSCcsharp.domain;
using log4net;
using log4net.Repository.Hierarchy;
using System.Data.SQLite;

namespace proiectSCcsharp.repository
{
    public class RepoAngajat : IRepoAngajat
    {
        private static readonly ILog logger = LogManager.GetLogger("RepoAngajat");

        IDictionary<String, string> props;

        public RepoAngajat(IDictionary<string, string> props)
        {
            logger.Info("Initializing RepoAngajat class");
            this.props = props;
        }

        public Angajat add(Angajat entity)
        {
            logger.InfoFormat("Saving angajat...", entity);

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "INSERT INTO Angajati(id_angajat, name, username, password) VALUES (@i, @n, @u, @p)";
                
                var paramId = command.CreateParameter();
                paramId.ParameterName = "@i";
                paramId.Value = null;
                command.Parameters.Add(paramId);

                var nameParameter = command.CreateParameter();
                nameParameter.ParameterName = "@n";
                nameParameter.Value = entity.name;
                command.Parameters.Add(nameParameter);

                var usernameParameter = command.CreateParameter();
                usernameParameter.ParameterName = "@u";
                usernameParameter.Value = entity.username;
                command.Parameters.Add(usernameParameter);
                
                var passParameter = command.CreateParameter();
                passParameter.ParameterName = "@p";
                passParameter.Value = entity.password;
                command.Parameters.Add(passParameter);

                var result = command.ExecuteNonQuery();

                return entity;
            }
        }
        public Angajat remove(long aLong)
        {
            logger.InfoFormat("Deleting angajat...", aLong);

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "DELETE FROM Angajati WHERE id_angajat=@i";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@i";
                idParameter.Value = aLong;
                command.Parameters.Add(idParameter);

                var result = command.ExecuteNonQuery();

                return findElement(aLong);
            }
        }
        public List<Angajat> getAll()
        {
            logger.InfoFormat("Finding all angajati...");

            List<Angajat> angajati = new List<Angajat>();

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Angajati";

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        long id = dataReader.GetInt64(0);
                        string name = dataReader.GetString(1);
                        string username = dataReader.GetString(2);
                        string password = dataReader.GetString(3);
                        Angajat angajat = new Angajat(id,name, username, password);
                        angajati.Add(angajat);
                    }
                }
            }

            return angajati;
        }

        public Angajat findElement(long id)
        {
            logger.InfoFormat("Entering findElement with value {0}", id);

            Angajat angajat = null;

            var connection = DBUtils.getConnection(props);

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "SELECT * FROM Angajati WHERE id_angajat = @i";

                var idParameter = command.CreateParameter();
                idParameter.ParameterName = "@i";
                idParameter.Value = id;
                command.Parameters.Add(idParameter);

                using (var dataReader = command.ExecuteReader())
                {
                    while (dataReader.Read())
                    {
                        long id2 = dataReader.GetInt64(0);
                        string name = dataReader.GetString(1);
                        string username = dataReader.GetString(2);
                        string password = dataReader.GetString(3);
                        angajat = new Angajat(id2, name, username, password);
                        logger.InfoFormat("Found angajat: {0}", angajat);
                        return angajat;
                    }
                }
            }
            return null;
        }
        public Angajat AuthenticateAngajat(string username, string password)
        {
            logger.InfoFormat("find a employee by username and password");
            Angajat angajat = null;
            var connection = DBUtils.getConnection(props);
            using (var command = new SQLiteCommand("select * from Angajati where username = @username and password = @password",
                      connection as SQLiteConnection))
            {
                command.Parameters.AddWithValue("@username", username);
                command.Parameters.AddWithValue("@password", password);
                using (var dataReader = command.ExecuteReader())
                    if (dataReader.Read())
                    {
                        long id2 = dataReader.GetInt64(0);
                        string name = dataReader.GetString(1);
                        string usernamee = dataReader.GetString(2);
                        string passwordd = dataReader.GetString(3);
                        angajat = new Angajat(id2,name, usernamee, passwordd);
                        logger.Info(angajat);
                        return angajat;
                    }
            }
            logger.InfoFormat("authentication failed");
            return null;
        }
    }
}

File: Anul 2/Semestrul 2/MPP/labs/Tema 6-proiect Java ORM/proiectServerClientJava/Persistance/src/main/java/proiectServerClientJava/persistence/repository/jdbc/RepoAngajat.java
package proiectServerClientJava.persistence.repository.jdbc;

import proiectServerClientJava.persistence.repository.IRepoAngajat;
import proiectServerClientJava.domain.Angajat;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoAngajat implements IRepoAngajat {
    private static final Logger logger= LogManager.getLogger();
    private final JDBCUtils jdbcUtils;

    public RepoAngajat(Properties properties) {
        logger.info("Initializing RepoAngajat with properties: {} ",properties);
        jdbcUtils=new JDBCUtils(properties);
    }

    @Override
    public Angajat add(Angajat entity) {
        logger.traceEntry("Saving angajat {}", entity);
        String query = "INSERT INTO Angajati(name,username,password) VALUES(?,?,?)";
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setString(1, entity.getName());
            statement.setString(2, entity.getUsername());
            statement.setString(3, entity.getPassword());

            int result=statement.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            throw new RuntimeException(e);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public Angajat remove(Integer aLong) {
        logger.traceEntry("Deleting angajat {}", aLong);
        String query = "DELETE FROM Angajati WHERE id_angajat = ?";

        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setInt(1, aLong);
            int result=statement.executeUpdate();
            logger.trace("Deleted {} instance", result);
        } catch (SQLException e) {
            logger.error(e);
            throw new RuntimeException(e);
        }
        logger.traceExit();
        return findElement(aLong);
    }

    @Override
    public Angajat findElement(Integer aLong) {
        logger.traceEntry();
        String query = "SELECT * FROM Angajati WHERE id_angajat=?";
        Angajat angajat = null;
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setInt(1, aLong);
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                Integer id = resultSet.getInt("id_angajat");
                String name = resultSet.getString("name");
                String username = resultSet.getString("username");
                String pass = resultSet.getString("password");
                angajat=new Angajat(id, name,username,pass);
            }
        } catch (SQLException e) {
            logger.error(e);
            e.printStackTrace();
        }
        logger.traceExit();
        return angajat;
    }

    @Override
    public List<Angajat> getAll() {
        logger.traceEntry();
        List<Angajat> angajati = new ArrayList<>();

        String query = "SELECT * from Angajati";
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query);
             ResultSet resultSet = statement.executeQuery()
        ) {
            while (resultSet.next()) {
                Integer id = resultSet.getInt("id_angajat");
                String name = resultSet.getString("name");
                String username = resultSet.getString("username");
                String pass = resultSet.getString("password");

                Angajat angajat=new Angajat(id,name,username,pass);
                angajati.add(angajat);
            }
        } catch (SQLException e) {
            logger.error(e);
            e.printStackTrace();
        }
        logger.traceExit();
        return angajati;
    }

    public Angajat authenticateAngajat(String username, String password) {
        logger.traceEntry();
        String query = "SELECT * FROM Angajati WHERE username=? and password=?";
        Angajat angajat = null;
        try (Connection connection = jdbcUtils.getConnection();
             PreparedStatement statement = connection.prepareStatement(query)
        ) {
            statement.setString(1, username);
            statement.setString(2, password);
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                Integer id=resultSet.getInt("id_angajat");
                String name = resultSet.getString("name");
                String usernm = resultSet.getString("username");
                String pass = resultSet.getString("password");
                angajat = new Angajat(id,name, usernm, pass);
            }
        } catch (SQLException e) {
            logger.error(e);
            e.printStackTrace();
        }
        logger.traceExit();
        return angajat;
    }
}
